src/persistent_state.rs

```diff
@@ -182,3 +186,43 @@ pub async fn save_session(
 
     Ok(())
 }
+
+/// Save the room panel when window is closed.
```

Comment:
this function itself doesn't have anything to do with the window being closed.
```suggestion
/// Save the current display state of the room panel to persistent storage.
```